src/js/letters/utils/validations.js

```diff
@@ -0,0 +1,56 @@
+import { isValidUSZipCode } from '../../common/utils/validations';
+
+const requiredMessage = 'Please enter a valid address'; // Change me! 
+
+/**
+ * Ensures the input isn't blank
+ */
+const requiredValidator = (input, fullAddress, message) => {
+  if (!input) {
+    return message || requiredMessage;
+  }
+
+  // Could return anything that isn't a string, really
+  return true;
+};
+
+
+/**
+ * These validation functions must return an error message string if the validation fails.
+ */
+export const addressOneValidations = [
+  (input, fullAddress) => requiredValidator(input, fullAddress, 'Please enter an address')
```

Comment:
How does this syntax work? How does it know what `input` and `fullAddress` are? This looks like magic!

Reply:
In `AddressSection`, `validateField()` passes the validation functions these params:
```js
errorMessage = validations[i](fullAddress[fieldName], fullAddress);
```

All validations functions have to have this signature to work.

Would it be helpful for me to write up some documentation in the comments about this?

Reply:
Ah okay, yes that helps!